pkg/controller/pxc/config.go

```diff
@@ -0,0 +1,301 @@
+	}
+	if result == controllerutil.OperationResultNone {
+		result = res
+	}
+
+	res, err = r.reconcileHAProxyConfigMap(ctx, cr)
+	if err != nil {
+		return result, errors.Wrap(err, "reconcile haproxy config map")
+	}
+	if result == controllerutil.OperationResultNone {
+		result = res
+	}
+
+	res, err = r.reconcileLogcollectorConfigMap(ctx, cr)
+	if err != nil {
+		return result, errors.Wrap(err, "reconcile logcollector config map")
+	}
+	if result == controllerutil.OperationResultNone {
+		result = res
+	}
+
+	_, err = r.reconcileHookScriptConfigMaps(ctx, cr)
+	if err != nil {
+		return result, errors.Wrap(err, "reconcile hookscript config maps")
+	}
+
+	return result, nil
+}
+
+func (r *ReconcilePerconaXtraDBCluster) reconcileAutotuneConfigMap(ctx context.Context, cr *api.PerconaXtraDBCluster) (controllerutil.OperationResult, error) {
+	autotuneCm := config.AutoTuneConfigMapName(cr.Name, "pxc")
+
+	_, ok := cr.Spec.PXC.Resources.Limits[corev1.ResourceMemory]
+	if !ok {
+		err := deleteConfigMapIfExists(ctx, r.client, cr, autotuneCm)
+		return controllerutil.OperationResultNone, errors.Wrap(err, "delete configmap")
+	}
+
+	configMap, err := config.NewAutoTuneConfigMap(cr, cr.Spec.PXC.Resources.Limits.Memory(), autotuneCm)
+	if err != nil {
+		return controllerutil.OperationResultNone, errors.Wrap(err, "new configmap")
+	}
+
+	err = k8s.SetControllerReference(cr, configMap, r.scheme)
+	if err != nil {
+		return controllerutil.OperationResultNone, errors.Wrap(err, "set controller ref")
+	}
+
+	res, err := createOrUpdateConfigmap(ctx, r.client, configMap)
+	if err != nil {
+		return res, errors.Wrap(err, "create or update configmap")
+	}
+
+	return res, nil
+}
+
+func (r *ReconcilePerconaXtraDBCluster) reconcileCustomConfigMap(ctx context.Context, cr *api.PerconaXtraDBCluster) (controllerutil.OperationResult, error) {
+	pxcConfigName := config.CustomConfigMapName(cr.Name, "pxc")
+
+	if cr.Spec.PXC.Configuration == "" {
+		err := deleteConfigMapIfExists(ctx, r.client, cr, pxcConfigName)
+		return controllerutil.OperationResultNone, errors.Wrap(err, "delete config map")
+	}
+
+	configMap := config.NewConfigMap(cr, pxcConfigName, "init.cnf", cr.Spec.PXC.Configuration)
+
+	err := k8s.SetControllerReference(cr, configMap, r.scheme)
+	if err != nil {
+		return controllerutil.OperationResultNone, errors.Wrap(err, "set controller ref")
+	}
+
+	res, err := createOrUpdateConfigmap(ctx, r.client, configMap)
+	if err != nil {
+		return res, errors.Wrap(err, "create or update config map")
+	}
+
+	return res, nil
+}
+
+func (r *ReconcilePerconaXtraDBCluster) reconcileHookScriptConfigMaps(ctx context.Context, cr *api.PerconaXtraDBCluster) (controllerutil.OperationResult, error) {
+	pxcHookScriptName := config.HookScriptConfigMapName(cr.Name, "pxc")
+	if cr.Spec.PXC != nil && cr.Spec.PXC.HookScript != "" {
+		err := r.createHookScriptConfigMap(ctx, cr, cr.Spec.PXC.PodSpec.HookScript, pxcHookScriptName)
```

Comment:
We are validating for `cr.Spec.PXC.HookScript != ""` but we are using `cr.Spec.PXC.PodSpec.HookScript`. At the end the same `HookScript` is used but let's be consistent and use one reference.